numpy/lib/function_base.py

```diff
@@ -356,7 +356,7 @@ def jhat(nbins):
         p_k = np.histogram(x, bins=nbins)[0] / n
         return (2 - (n + 1) * p_k.dot(p_k)) / hh
 
-    nbins = min(range(1, max(n // 4, 2)), key=jhat)
+    nbins = min(range(1, max(100, int(np.sqrt(n)))), key=jhat)
```

Comment:
`max` was my idea, with the thought that it's OK to do extra work for the sake of a more exhaustive search on small datasets, because for small `n` the square-root rule may not be a strict upper bound. But I agree, 100 is probably too large.

The other non-data dependent rule we have is the Rice Rule. It suggests more bins than the square root rule up to `n=64` (8 bins with both). So at the very least we want the constant to be 8 or larger. Or we could explicitly use the Rice Rule here. Either we definitely want a comment explaining the choice ("search up to the larger of the Rice or square root rules").

It would be nice if we had some empirical test suite of distributions to try these rules on. It's hard to guess how much of a margin we need without trying this out on some real data...